src/utils/clone.ts

```diff
@@ -0,0 +1,41 @@
+import { isPlainObject } from './index'
+
+export const clone = <T>(obj: T): T => {
+  if (isPlainObject(obj)) {
```

Comment:
Does this check imply that object that are instances of classes don't work with this utility? Does this behave as expected when provided an instance of Response? How about objects that have properties that are instances of, say, Request?

Reply:
It's probably working fine right now since we only clone headers and bodies which are always plain objects, but we might want to at least leave a comment so the next person doesn't fall on the trap of cloning a whole Response.

Reply:
Added case for instance of class